### ISSUE TITLE
Reverse proxy works and page loads, but PiKVM doesn't connect

### ISSUE THREAD

**Issue by chriexpe** [pikvm/pikvm#1452]
This is a weird issue, it used to work just fine on this setup with reverse proxy using NGINX-PM (+ cloudflare for DNS).
The interface does load, I can login and access console (that just display "Reconnect") and KVM that only shows it's connecting (that orange Ethernet icon).

![image](https://github.com/user-attachments/assets/93960eed-29bb-4985-ab82-22e99a957be5)

![image](https://github.com/user-attachments/assets/51799014-5fdf-4a63-97b0-214556ae5dd6)

NGINX-PM Settings:
![image](https://github.com/user-attachments/assets/19f505eb-69f3-42d0-a49a-882e17508fb7)
![image](https://github.com/user-attachments/assets/f3b2d0a3-d353-4b66-be8e-92ee0df5fac7)

 - OS: All devices I tried, Android and Windows
 - Browser: Firefox, Edge, also both on mobile data

**PiKVM info:**
  - Raspberry Pi 4b 8Gb
  - PiKVM platform: v3 Hat
  - KVMD version: `kvmd 4.32-1 | kvmd-fan 0.31-1 | kvmd-platform-v3-hdmi-rpi4 4.32-1 | kvmd-webterm 0.50-1`
  - uStreamer version: `6.21-1`
  - Linux kernel: `6.6.45-8-rpi #1 SMP Wed Oct  2 19:56:38 UTC 2024 armv7l GNU/Linux`
  
I do have other things that are reverse proxied and they all work fine, even a WebRTC NVR.
Yes using local IP it works perfectly.

**Comment by mdevaev** [comment]
Hello. Some URLs require websockets passing, so you need to enable it on your proxy.

**Comment by chriexpe** [comment]
> Check JS browser console for errors.

![image](https://github.com/user-attachments/assets/74c153be-b8fa-4905-8c07-a0c44e28e728)

**Comment by mdevaev** [comment]
It looks like this is a proxy problem. It's breaking websocket, as I thought.

**Comment by chriexpe** [comment]
> It looks like this is a proxy problem. It's breaking websocket, as I thought.

Indeed, later I found out that Authentik's custom configuration for NGINX was missing some header, after adding it everything is working again, thanks!

For anyone that happens to stuble on this, just add `proxy_set_header Upgrade $http_upgrade;` below "location / {"
Like this:
```
location / {
    # Put your proxy_pass to your application here
    proxy_pass          $forward_scheme://$server:$port;
    # Set any other headers your application might need
    # proxy_set_header Host $host;
    # proxy_set_header ...
```

**Comment by ljo123** [comment]
> > It looks like this is a proxy problem. It's breaking websocket, as I thought.
> 
> Indeed, later I found out that Authentik's custom configuration for NGINX was missing some header, after adding it everything is working again, thanks!
> 
> For anyone that happens to stuble on this, just add `proxy_set_header Upgrade $http_upgrade;` below "location / {" Like this:
> 
> ```
> location / {
>     # Put your proxy_pass to your application here
>     proxy_pass          $forward_scheme://$server:$port;
>     # Set any other headers your application might need
>     proxy_set_header Upgrade $http_upgrade;
>     # proxy_set_header Host $host;
>     # proxy_set_header ...
> ```

Thanks! I just updated PiKVM and got this error. Everything had been working fine previously. Can you share a bit more detail on your NPM custom location config?

**Comment by ljo123** [comment]
Thanks but my issue is something different. Everything was working but then i ran the pikvm update function and now I have the error. I can connect directly to the local IP and it works but through NPM it won't connect to the remote device.
